media-source-respec.html

```diff
@@ -653,29 +653,29 @@
         <section id="mediasource-attach">
           <h4>Attaching to a media element</h4>
           <p> A <a>MediaSource</a> object can be attached to a media element by assigning a <a def-id="MediaSource-object-URL"></a> to the media element <a def-id="media-src"></a> attribute or the src attribute of a &lt;source&gt; inside a media element. A <a def-id="MediaSource-object-URL"></a> is created by passing a MediaSource object to <a def-id="createObjectURL"></a>.</p>
-          <p>If the <a def-id="resource-fetch-algorithm"></a> absolute URL matches the MediaSource object URL, ignore any <a def-id="preload"></a> attribute of the media element, skip any optional steps for when <a def-id="preload"></a> equals <a def-id="preload-none"></a>, and run the following steps right before the <a def-id="perform-potentially-cors-enabled-fetch"></a>
-            step in the <a def-id="resource-fetch-algorithm"></a>.</p>
 
+          <p>If the <a def-id="resource-fetch-algorithm"></a> was invoked with a media provider object that is a MediaSource object or a URL record whose object is a MediaSource object, then let mode be local, skip the first step in the <a def-id="resource-fetch-algorithm"></a> which may otherwise set mode to remote, and run the following steps at the beginning of the <a def-id="Otherwise-mode-is-local"></a> section of the <a def-id="resource-fetch-algorithm"></a>.
```

Comment:
Moving comment to respec file:

This follows the model of the previous section contents, but feels awkward. The goal is to set the mode to local, skip the first resource fetch algorithm step (which could set mode to remote), add some steps that are MediaSource specific, and modify the "Otherwise (mode is local)" steps to fit.

Since it follows the previous structure here, it's probably okay. I'd be tempted to say something more like:

`If...then let mode be local, skip the first step in the resource fetch algorithm (which may otherwise set the mode to remote, and add the steps and clarifications below to the "Otherwise (mode is local)" section of the resource fetch algorithm.`

This might seem like wordsmithing. I'm reacting a bit to the "before" aspect of the steps, when they are partly before, and partly alterations of the steps themselves.